packages/protocol/contracts/governance/LockedGold.sol

```diff
@@ -5,777 +5,437 @@ import "openzeppelin-solidity/contracts/math/SafeMath.sol";
+   * @param value The amount by which to decrement.
+   * @dev Can only be called by the registered "Election" smart contract.
    */
-  function increaseNoticePeriod(
-    uint256 value,
-    uint256 noticePeriod,
-    uint256 increase
+  function decrementNonvotingAccountBalance(
+    address account,
+    uint256 value
   )
     external
-    nonReentrant
-    returns (uint256)
+    onlyRegisteredContract(ELECTION_REGISTRY_ID)
   {
-    require(isAccount(msg.sender) && !isVoting(msg.sender));
-    // _redeemRewards(msg.sender);
-    require(value > 0 && increase > 0);
-    Account storage account = accounts[msg.sender];
-    Commitment storage locked = account.commitments.locked[noticePeriod];
-    require(locked.value >= value);
-    updateLockedCommitment(account, uint256(locked.value).sub(value), noticePeriod);
-    uint256 increasedNoticePeriod = noticePeriod.add(increase);
-    uint256 increasedValue = account.commitments.locked[increasedNoticePeriod].value;
-    updateLockedCommitment(account, increasedValue.add(value), increasedNoticePeriod);
-    emit NoticePeriodIncreased(msg.sender, value, noticePeriod, increase);
-    return account.weight;
+    _decrementNonvotingAccountBalance(account, value);
   }
 
   /**
-   * @notice Returns whether or not an account's voting power is frozen.
-   * @param account The address of the account.
-   * @return Whether or not the account's voting power is frozen.
-   * @dev Frozen accounts can retract existing votes but not make future votes.
+   * @notice Increments the non-voting balance for an account.
+   * @param account The account whose non-voting balance should be incremented.
+   * @param value The amount by which to increment.
    */
-  function isVotingFrozen(address account) external view returns (bool) {
-    return accounts[account].votingFrozen;
+  function _incrementNonvotingAccountBalance(address account, uint256 value) private {
+    accounts[account].balances.nonvoting = accounts[account].balances.nonvoting.add(value);
+    totalNonvoting = totalNonvoting.add(value);
   }
 
   /**
-   * @notice Returns the timestamp of the last time the account redeemed block rewards.
-   * @param _account The address of the account.
-   * @return The timestamp of the last time `_account` redeemed block rewards.
+   * @notice Decrements the non-voting balance for an account.
+   * @param account The account whose non-voting balance should be decremented.
+   * @param value The amount by which to decrement.
    */
-  function getRewardsLastRedeemed(address _account) external view returns (uint96) {
-    Account storage account = accounts[_account];
-    return account.rewardsLastRedeemed;
-  }
-
-  function isValidating(address validator) external view returns (bool) {
-    IValidators validators = IValidators(registry.getAddressFor(VALIDATORS_REGISTRY_ID));
-    return validators.isValidating(validator);
+  function _decrementNonvotingAccountBalance(address account, uint256 value) private {
```

Comment:
Looks like the new version of solhint has this check disabled by default. I'll reorder these in a follow-up PR to avoid merge conflicts in asaj/pos-2

Reply:
We can disable that if we want. Not sure what we prefer.

Reply:
I think it would be good to enable, the function ordering in my PR is a bit nonsensical and hard to read. Only keeping it this way to minimize merge conflicts with asaj/pos-2